index.bs

```diff
@@ -485,6 +510,85 @@ spec:csp3; type:grammar; text:base64-value
+  enforce a policy regarding the integrity metadata requirements on all the subresources it
+  loads of certain <a>destinations</a>.
+
+  The headers' value is a <a>Dictionary</a> [[RFC9651]], with every member-value being an
+  <a>inner list</a> of <a>tokens</a>.
+  
+  An <dfn>integrity policy struct</dfn>, is a <a>struct</a> that contains the following:
+    * <dfn>enforcement</dfn>, a string which is the empty string or "block". Initially "block".
+    * <dfn>source</dfn>, a string which is the empty string or "inline". Initially "inline".
+    * <dfn>destination</dfn>, a string which is the empty string or "script". Initially the empty string.
+    * <dfn>endpoints</dfn>, a <a>list</a> of strings, initially empty.
+
+  When <dfn>processing an integrity policy</dfn>, with |input|, do the following:
+  1) Let |integrity policy| be a new <a>integrity policy struct</a>.
+  1) Let |dictionary| be the result of <a>parsing a dictionary</a> with |input|.
+  1) If |dictionary|['enforcement'] exists:
+    1) If its value is "block", set |integrity policy|'s <a>enforcement</a> to "block".
+    1) Otherwise, set |integrity policy|'s <a>enforcement</a> to the empty string.
+  1) If |dictionary|['sources'] exists:
+    1) If its value <a>contains</a> "inline", set |integrity policy|'s <a>source</a> to "inline".
+    1) Otherwise, set |integrity policy|'s <a>source</a> to the empty string.
+  1) If |dictionary|['destinations'] exists:
+    1) If its value <a>contains</a> "script", set |integrity policy|'s <a>destination</a> to "script".
+  1) if |dictionary|['endpoints'] exists:
+    1) Set |integrity policy|'s <a>endpoints</a> to |dictionary|['endpoints'].
+  1) return |integrity policy|
+  
+  ### Parse Integrity-Policy headers ### {#parse-integrity-policy-headers-section}
+  To <dfn>parse Integrity-Policy headers</dfn>, given a <a>Response</a> |response|
+  and a <a>policy container</a> |container|, do the following:
+  1) Let |headers| be |response|'s <a>header list</a>.
+  1) If |headers| <a>contains</a> a <a>header</a> whose <a>header name</a> is "Integrity-Policy",
+     let |container|'s <a>integrity policy</a> be the result of running <a>processing an integrity policy</a>
+     with the corresponding <a>header value</a>.
+  1) If |headers| <a>contains</a> a <a>header</a> whose <a>header name</a> is "Integrity-Policy-Report-Only",
+     let |container|'s <a>report only integrity policy</a> be the result of running <a>processing an integrity
+     policy</a> with the corresponding <a>header value</a>.
+  
+  ### Should request be blocked by Integrity Policy ### {#should-request-be-blocked-by-integrity-policy-section}
+  To determine <dfn>should request be blocked by integrity policy</dfn>, given a <a>request</a> |request|,
+  do the following:
+  1) Let |policy container| be |request|'s <a>policy container</a>.
+  1) if |request|'s <a>integrity metadata</a> is not empty, return "Allowed".
```

Comment:
(This is correct for now, but we're going to have to change that for situations where integrity is not inline. I think I had a thought, but maybe this is fine as is?)

Reply:
I agree we'd need to change this once Integrity-Policy evolves, but those changes would also include changes to "integrity metadata" (e.g. to include its source). I prefer we'd tackle that when we need it.